exchange/auction.go

```diff
@@ -47,6 +48,52 @@ func (d *DebugLog) BuildCacheString() {
 	d.CacheString = fmt.Sprintf("%s<Log>%s%s%s</Log>", xml.Header, d.Data.Request, d.Data.Headers, d.Data.Response)
 }
 
+func (d *DebugLog) PutDebugLogError(cache prebid_cache_client.Client, errors []error) error {
+	if len(d.Data.Response) == 0 && len(errors) == 0 {
+		d.Data.Response = "No response or errors created"
+	}
+
+	if len(errors) > 0 {
+		errStrings := []string{}
+		for _, err := range errors {
+			errStrings = append(errStrings, err.Error())
+		}
+		d.Data.Response = fmt.Sprintf("%s\nErrors:\n%s", d.Data.Response, strings.Join(errStrings, "\n"))
+	}
+
+	d.BuildCacheString()
+
+	if len(d.CacheKey) == 0 {
+		rawUUID, err := uuid.NewV4()
+		if err != nil {
+			return err
+		}
+		d.CacheKey = rawUUID.String()
+	}
+
+	data, err := json.Marshal(d.CacheString)
+	if err != nil {
+		return err
+	}
+
+	toCache := []prebid_cache_client.Cacheable{
+		{
+			Type:       d.CacheType,
+			Data:       data,
+			TTLSeconds: d.TTL,
+			Key:        "log_" + d.CacheKey,
+		},
+	}
+
+	if cache != nil {
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(time.Duration(100)*time.Millisecond))
```

Comment:
Again, I understand that this is not a new change in your PR but curious as to why we're using 100ms as the timeout for the cache request? I mean how did we determine that 100ms is a good value here?

Reply:
100ms was kind of chosen as just a safe starting point until this could be a config value. I will check into adding this.

Reply:
Added a config value for the cache deadline. Let me know if this improves this portion.